<a href="https://colab.research.google.com/github/qitinnstud/myfirstexperience/blob/For_homework/For_homework1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Импорт библиотеки pandas

In [2]:
import pandas as pd
#from google.colab import files
#uploaded = files.upload()
#import io
#df = pd.read_csv(io.BytesIO(uploaded['Weather Data.csv']))

Монтирую google drive

In [4]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


Читаю DataFrame в переменную df0 и вывожу первые 11 строк

In [49]:
df0=pd.read_csv("drive/MyDrive/Weather_Data.csv");
df0.head(11)

Date/Time  Temp_C  Dew Point Temp_C  Rel Hum_%  Wind Speed_km/h  \
0    1/1/2012 0:00    -1.8              -3.9       86.0              4.0   
1    1/1/2012 1:00    -1.8              -3.7       87.0              4.0   
2    1/1/2012 2:00    -1.8              -3.4       89.0              7.0   
3    1/1/2012 3:00    -1.5              -3.2       88.0              6.0   
4    1/1/2012 4:00    -1.5              -3.3       88.0              7.0   
5    1/1/2012 5:00    -1.4              -3.3       87.0              9.0   
6    1/1/2012 6:00    -1.5              -3.1       89.0              7.0   
7    1/1/2012 7:00    -1.4              -3.6       85.0              7.0   
8    1/1/2012 8:00    -1.4              -3.6       85.0              9.0   
9    1/1/2012 9:00    -1.3              -3.1       88.0             15.0   
10  1/1/2012 10:00    -1.0              -2.3       91.0              9.0   

    Visibility_km  Press_kPa               Weather  
0             8.0     101.24                   Fog  
1             8.0     101.24                   Fog  
2             4.0     101.26  Freezing Drizzle,Fog  
3             4.0     101.27  Freezing Drizzle,Fog  
4             4.8     101.23                   Fog  
5             6.4     101.27                   Fog  
6             6.4     101.29                   Fog  
7             8.0     101.26                   Fog  
8             8.0     101.23                   Fog  
9             4.0     101.20                   Fog  
10            1.2     101.15                   Fog

Читаю DataFrame c GitHub в переменную df и вывожу несколько строк

In [42]:
df.to_csv("drive/MyDrive/Weather_Data1.csv",sep='|',index=False)

In [43]:
url='https://raw.githubusercontent.com/qitinnstud/myfirstexperience/For_homework/Weather_Data.raw'
df = pd.read_csv(url, sep=',')
df.head()

Date/Time  Temp_C  Dew Point Temp_C  Rel Hum_%  Wind Speed_km/h  \
0  1/1/2012 0:00    -1.8              -3.9       86.0              4.0   
1  1/1/2012 1:00    -1.8              -3.7       87.0              4.0   
2  1/1/2012 2:00    -1.8              -3.4       89.0              7.0   
3  1/1/2012 3:00    -1.5              -3.2       88.0              6.0   
4  1/1/2012 4:00    -1.5              -3.3       88.0              7.0   

   Visibility_km  Press_kPa               Weather  
0            8.0     101.24                   Fog  
1            8.0     101.24                   Fog  
2            4.0     101.26  Freezing Drizzle,Fog  
3            4.0     101.27  Freezing Drizzle,Fog  
4            4.8     101.23                   Fog

Запрос инфо датафрейма. Вывод информации о колонках (заголовках) и типа данных в них:

In [51]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8784 entries, 0 to 8783
Data columns (total 8 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Date/Time         8784 non-null   object 
 1   Temp_C            8779 non-null   float64
 2   Dew Point Temp_C  8776 non-null   float64
 3   Rel Hum_%         8782 non-null   float64
 4   Wind Speed_km/h   8782 non-null   float64
 5   Visibility_km     8780 non-null   float64
 6   Press_kPa         8779 non-null   float64
 7   Weather           8778 non-null   object 
dtypes: float64(6), object(2)
memory usage: 549.1+ KB


Запрос статистики датафрейма (число строк, среднее, СТО, мин, макс. и квартили):

In [53]:
df.describe()

Temp_C  Dew Point Temp_C    Rel Hum_%  Wind Speed_km/h  \
count  8779.000000       8776.000000  8782.000000      8782.000000   
mean      8.799168          2.560472    67.422455        14.944432   
std      11.687429         10.879407    16.925787         8.689410   
min     -23.300000        -28.500000    18.000000         0.000000   
25%       0.100000         -5.900000    56.000000         9.000000   
50%       9.300000          3.300000    68.000000        13.000000   
75%      18.800000         11.800000    81.000000        20.000000   
max      33.000000         24.400000   100.000000        83.000000   

       Visibility_km    Press_kPa  
count    8780.000000  8779.000000  
mean       27.675773   101.051455  
std        12.643544     0.844127  
min         0.200000    97.520000  
25%        24.100000   100.560000  
50%        25.000000   101.070000  
75%        25.000000   101.590000  
max       100.690000   103.650000

включая стр. тип object:

In [55]:
df.describe(include='object')

Date/Time       Weather
count            8784          8778
unique           8784            50
top     1/1/2012 0:00  Mainly Clear
freq                1          2105

Дата-время представлиены типом 'object'. "Преимущественно ясно" встречается чаще - 2105 вхождений.

In [56]:
df['Date/Time'] = pd.to_datetime(df['Date/Time'])

In [60]:
df.describe(include='datetime')

<ipython-input-60-41adad5e0539>:1: FutureWarning: Treating datetime data as categorical rather than numeric in `.describe` is deprecated and will be removed in a future version of pandas. Specify `datetime_is_numeric=True` to silence this warning and adopt the future behavior now.
  df.describe(include='datetime')


Date/Time
count                  8784
unique                 8784
top     2012-01-01 00:00:00
freq                      1
first   2012-01-01 00:00:00
last    2012-12-31 23:00:00

In [61]:
df.isna().sum()

Date/Time           0
Temp_C              5
Dew Point Temp_C    8
Rel Hum_%           2
Wind Speed_km/h     2
Visibility_km       4
Press_kPa           5
Weather             6
dtype: int64